lib/os/crc32c_sw.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2021 Workaround GmbH.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/crc.h>
+
+/* crc table generated from polynomial 0x1EDC6F41UL (Castagnoli) */
+static const uint32_t crc32c_table[256] = {
```

Comment:
You need to implement this crc func without this large lookup table to save flash, just like how it is done for crc32.

Reply:
But, isn't that a hit on performance? For the application where i am using it, performance is critical. We have to calculate the CRC for tens of MBs of data. And we have lots of flash space left.

Reply:
That is your use case, but there are other use cases too. Currently all the other crc functions are semi-optimized for smaller lookup tables or there is no lookup table at all. For example `crc32_ieee_update()` does have small lookup table of 64 bytes.

Reply:
Made changes to use a smaller lookup of only 16 bytes.